ee/clickhouse/client.py

```diff
@@ -133,26 +134,36 @@ def sync_execute(query, args=None, settings=None, with_column_types=False):
         with ch_pool.get_client() as client:
             start_time = perf_counter()
             tags = {}
-            if app_settings.SHELL_PLUS_PRINT_SQL:
-                print()
-                print(format_sql(query, args))
 
-            parameterized_sql, tags = _annotate_tagged_query(query, args)
+            # NOTE: `client.execute` would normally handle substitution, but
+            # because we want to strip the comments to make it easier to copy
+            # and past queries from the `system.query_log` easily with metabase
+            # (metabase doesn't show new lines, so with comments, you can't get
+            # a working query without exporting to csv or similar), we need to
+            # do it manually.
+            #
+            # We only want to try to substitue for SELECT queries, which
+            # clickhouse_driver at this moment in time decides based on the
+            # below predicate.
+            if isinstance(args, (list, tuple, types.GeneratorType)):
+                rendered_sql = query
+            else:
+                rendered_sql = client.substitute_params(query, args or {})
+            formatted_sql = sqlparse.format(rendered_sql, strip_comments=True)
+
+            # Adds in a /* */ so we can look in clickhouses `system.query_log`
+            # to easily marry up to the generating code.
+            annotated_sql, tags = _annotate_tagged_query(formatted_sql, args)
```

Comment:
Does this code block with ~20 lines of code belong in this function? When I was reading this function I'd not care about it.